src/samplers/resampler.jl

```diff
@@ -0,0 +1,53 @@
+struct WeightedResampler{F<:VariateForm, S<:ValueSupport} <: Sampleable{F, S}
+    obs::AbstractArray
+    wv::AbstractWeights
+end
+
+"""
```

Comment:
This constructor could be the default one, and the docstring moved to the struct definition

Reply:
I'm not sure I know what you mean by "default". Are you suggesting that it should be an inner constructor or just that the docstring should be attached to the type?

Reply:
sorry, yes that it should be attached to the type.